oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/MappedJwtClaimSetConverter.java

```diff
@@ -0,0 +1,232 @@
+
+		Map<String, Converter<Object, ?>> attributeNameToConverter = new HashMap<>();
+		attributeNameToConverter.put(JwtClaimNames.AUD, AUDIENCE_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.EXP, TEMPORAL_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.IAT, TEMPORAL_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.ISS, ISSUER_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.JTI, STRING_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.NBF, TEMPORAL_CONVERTER);
+		attributeNameToConverter.put(JwtClaimNames.SUB, STRING_CONVERTER);
+		attributeNameToConverter.putAll(attributeConverters);
+
+		return new MappedJwtClaimSetConverter(attributeNameToConverter);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Map<String, Object> convert(Map<String, Object> claims) {
+		Assert.notNull(claims, "claims cannot be null");
+
+		Map<String, Object> mappedClaims = new HashMap<>(claims);
+
+		for (Map.Entry<String, Converter<Object, ?>> entry : this.attributeConverters.entrySet()) {
+			Converter<Object, ?> converter = this.attributeConverters.get(entry.getKey());
+			if (converter != null) {
+				Object claim = claims.get(entry.getKey());
+				Object mappedClaim = converter.convert(claim);
+				put(mappedClaims, entry.getKey(), mappedClaim);
+			}
+		}
+
+		Instant issuedAt = (Instant) mappedClaims.get(JwtClaimNames.IAT);
+		Instant expiresAt = (Instant) mappedClaims.get(JwtClaimNames.EXP);
+		if (issuedAt == null && expiresAt != null) {
+			mappedClaims.put(JwtClaimNames.IAT, expiresAt.minusSeconds(1));
+		}
+
+		return mappedClaims;
+	}
+
+	private static class AudienceConverter implements Converter<Object, Collection<String>> {
+
+		@Override
+		public Collection<String> convert(Object audience) {
+			if (audience == null) {
+				return null;
+			}
+
+			if (audience instanceof Collection) {
+				return ((Collection<?>) audience).stream()
+						.filter(Objects::nonNull)
+						.map(Objects::toString)
+						.collect(Collectors.toList());
+			}
+
+			return Arrays.asList(audience.toString());
+		}
+	}
+
+	private static class IssuerConverter implements Converter<Object, URL> {
+
+		@Override
+		public URL convert(Object issuer) {
+			if (issuer == null) {
+				return null;
+			}
+
+			if (issuer instanceof URL) {
+				return (URL) issuer;
+			}
+
+			if (issuer instanceof URI) {
+				return toUrl((URI) issuer);
+			}
+
+			return toUrl(issuer.toString());
+		}
+
+		private URL toUrl(String issuer) {
+			try {
+				return new URL(issuer);
+			} catch (MalformedURLException e) {
+				throw new IllegalStateException("Could not coerce iss value to URL");
+			}
+		}
+
+		private URL toUrl(URI issuer) {
+			try {
+				return issuer.toURL();
+			} catch (MalformedURLException e) {
+				throw new IllegalStateException("Could not coerce iss value to URL");
+			}
+		}
+	}
+
+	private static class TemporalConverter implements Converter<Object, Instant> {
```

Comment:
Is there a reason this is called TemporalConverter? It converts to an Instant which is what the other converters are named after.